rosette/src/Vm.cc

```diff
@@ -512,7 +515,7 @@ void VirtualMachine::handleSignal() {
     if (sigvec & sigmask(SIGINT)) {
         printf("aborting...\n");
         reset();                   /* clears sigvec */
-        (void)sigsetmask(oldmask); /* enable interrupts */
+        (void)sigprocmask(SIG_SETMASK, &oldmask, NULL); /* enable interrupts */
         return;
     }
 
```

Comment:
Please add a TODO here and then file a bug to get rid of the `goto`. Holy hell.

Reply:
It doesn't appear that this is the actual signal handler.  The actual handler is in Interrupt.cc RosetteSignalHandler()  which calls vm->acceptSignal with the signal raised which in turn stores it in the sigvec variable.

handleSignal() is called from what I think is the main loop in VirtualMachine::getNextStrand() .